src/http/status_code.cr

```diff
@@ -0,0 +1,162 @@
+  def self.success?(status_code : Int)
+    (200..299).includes?(status_code)
+  end
+
+  # Returns `true` if the response status code is between 300 and 399.
+  def self.redirection?(status_code : Int)
+    (300..399).includes?(status_code)
+  end
+
+  # Returns `true` if the response status code is between 400 and 499.
+  def self.client_error?(status_code : Int)
+    (400..499).includes?(status_code)
+  end
+
+  # Returns `true` if the response status code is between 500 and 599.
+  def self.server_error?(status_code : Int)
+    (500..599).includes?(status_code)
+  end
+
+  # Returns the default status message of the given HTTP status code.
+  def self.default_message_for(status_code : Int) : String
+    case status_code
+    when 100 then "Continue"
+    when 101 then "Switching Protocols"
+    when 102 then "Processing"
+    when 200 then "OK"
+    when 201 then "Created"
+    when 202 then "Accepted"
+    when 203 then "Non-Authoritative Information"
+    when 204 then "No Content"
+    when 205 then "Reset Content"
+    when 206 then "Partial Content"
+    when 207 then "Multi-Status"
+    when 208 then "Already Reported"
+    when 226 then "IM Used"
+    when 300 then "Multiple Choices"
+    when 301 then "Moved Permanently"
+    when 302 then "Found"
+    when 303 then "See Other"
+    when 304 then "Not Modified"
+    when 305 then "Use Proxy"
+    when 307 then "Temporary Redirect"
+    when 308 then "Permanent Redirect"
+    when 400 then "Bad Request"
+    when 401 then "Unauthorized"
+    when 402 then "Payment Required"
+    when 403 then "Forbidden"
+    when 404 then "Not Found"
+    when 405 then "Method Not Allowed"
+    when 406 then "Not Acceptable"
+    when 407 then "Proxy Authentication Required"
+    when 408 then "Request Timeout"
+    when 409 then "Conflict"
+    when 410 then "Gone"
+    when 411 then "Length Required"
+    when 412 then "Precondition Failed"
+    when 413 then "Payload Too Large"
+    when 414 then "URI Too Long"
+    when 415 then "Unsupported Media Type"
+    when 416 then "Range Not Satisfiable"
+    when 417 then "Expectation Failed"
+    when 421 then "Misdirected Request"
+    when 422 then "Unprocessable Entity"
+    when 423 then "Locked"
+    when 424 then "Failed Dependency"
+    when 426 then "Upgrade Required"
+    when 428 then "Precondition Required"
+    when 429 then "Too Many Requests"
+    when 431 then "Request Header Fields Too Large"
+    when 451 then "Unavailable For Legal Reasons"
+    when 500 then "Internal Server Error"
+    when 501 then "Not Implemented"
+    when 502 then "Bad Gateway"
+    when 503 then "Service Unavailable"
+    when 504 then "Gateway Timeout"
+    when 505 then "HTTP Version Not Supported"
+    when 506 then "Variant Also Negotiates"
+    when 507 then "Insufficient Storage"
+    when 508 then "Loop Detected"
+    when 510 then "Not Extended"
+    when 511 then "Network Authentication Required"
+    else          ""
```

Comment:
Is this really the best solution? Maybe `"Unknown Status Code"` or something?

Reply:
When you'll use an enum, this can be changed to a macro: `{% for member in @type.constants %}`
You'll have to replace `_` by a space.
I don't know for the dash, maybe a double underscore `__`

Reply:
Thanks for that tip - hadn't even considered macros.

Looking at the Crystal docs for enums it looked like the keys would be camelcased, like `HTTP::StatusCode::NetworkAuthenticationRequired`. I was hoping there might be some sort of `humanize` method that would give me `Network authentication required` back but doesn't seem like it. 

From the looks of it it's not particularly important that the string returned here matches the spec, but that it gives a human representation of it - so I think losing some capitalisation or a hyphen might not be an issue.

Reply:
I don't think this should be implemented with macros. The list is completely fine. The returned string message should match the official description.

Enum values need to start with an upper case letter, as does any constant. But the name can be camel cased or snake cased and there are instances of both styles in the stdlib.